### DISCUSSION TITLE
Podman Support

### DISCUSSION THREAD

**Discussion by sudo-kraken** [mag37/dockcheck#108] [Ideas]
Hey, I recently helped with the pushover integration and have just migrated from docker to podman and I love this tool and would very much like to help to get this compatible with podman as well if that is something you find interesting for this project?

**Comment by mag37** [comment]
I'm happy for all help!

Podman support would be awesome if it's just some minor tweaks and I'd be happy to discuss and test this further.
I'll convert this to a discussion instead and we can brainstorm some findings and situations.
Thank you for the interest!

**Comment by sudo-kraken** [comment]
Apologies I only just saw you converted this over, the first thing we should probably think about is the extent of podman support you want to add, I.e. just podman rootless containers, or include rootful and do you want to support quadlet (my vote on the latter is yes)

**Comment by sudo-kraken** [comment]
Hey, for now I have forked this and it is available here, I have it working for my use case, and I have tried to make it compatible with most podman installs but its working great for me with Quadlet!

https://github.com/sudo-kraken/podcheck

**Comment by mag37** [comment]
Oh and I completely missed your reply a few days back.

Impressive work! I've not had time to inspect the code yet but great initiative!

And I'm glad you chose to focus on Quadlets too as that's how I run my pods too 😁

I'll add a link from Dockcheck to your project - because I guess it's rewritten a bunch and you'll keep maintaining it?

**Reply by sudo-kraken** [reply]
I certainly will, ideally I could do with some people testing and raising issues as I can't test every deployment type. I will also try to keep up with new features in dockcheck and port them over and any additional notification types you add 👍

**Reply by mag37** [reply]
Wonderful! I'll see if I can test it out on my Podman-rig, though I've set that up with auto-updates.
I wrote a blogpost quite recently on how I've started with podman: https://mag37.org/posts/guide_podman_quadlets/ 

Did a quick glance at your changes - and you've done a bunch of sensible formatting changes too, with comments and things. I might feed them back into DockCheck at some point if you don't mind.

Will read the code changes in depth when I've got the time, just stumbled upon the change from `builtin type -P xyz` to `command -v xyz` - any particular reason?

**Reply by sudo-kraken** [reply]
I don't mind at all. The primary reason for changing from builtin type -P xyz to command -v xyz is to enhance the portability and compatibility of the script across different shells. By using command -v, it just means the sscript behaves consistently and reduces the likelihood of errors due to shell-specific differences.

**Reply by sudo-kraken** [reply]
Also, excellent blog post!

**Reply by mag37** [reply]
Ah that's great - I've tried making it more generic/portable where I knew how or people reported issues. Didn't know about that case - thank you for the info!

I'd happily use your improvements and discuss any changes to Dockcheck that would ease your work with Podcheck.


> Also, excellent blog post!

Thank you, just started so havnt really explored it as much as I'd like 🙂 will endeavour further when time allows!

**Reply by sudo-kraken** [reply]
Sounds great to me 😄 just need to get it tested by more people and keep an open discussion going, we will need to sort out a way to communicate on changes and potential improvements that can be utilised in both tools.

**Reply by sudo-kraken** [reply]
I have sent you an invite to be a collaborator on Podcheck, so if you see something you can contribute easily.

**Reply by mag37** [reply]
I just ran it through my stack of 19 containers and it worked great! Nice job!
(Seems like my auto-update setup need some investigation..)

If you use Matrix you could add me there for further communication. `@mag37:matrix.org`
